### DISCUSSION TITLE
Running KAPE at scale

### DISCUSSION THREAD

**Discussion by antmar904** [EricZimmerman/KapeFiles#658] [General]
Hi

I just wanted to know how some people are running KAPE at scale.  In other words how are you running KAPE on hundreds of endpoints, collecting the data then running the modules against your collections?

I am using our XDR tool to download KAPE to local disk. run KAPE, zip up the collection then upload to Azure.  I then download those collections to my test box.  Can I then loop through those collections with PowerShell and run KAPE modules against them in parallel or one by one or is there a better way?

**Comment by oighen** [comment]
Velociraptor! Watch the latest Eric Capuano YouTube video, or go to the documentation for Velociraptor. You need only 1 server for 5-10k endpoints. More than that? There are ways. You will be amazed.

**Reply by antmar904** [reply]
So your running KAPE via Velociraptor?

**Comment by EricZimmerman** [comment]
There's no real right or wrong way to do this. I purposely did not build in a mechanism for deploying or calling things at scale because everybody does things very differently Even inside the same company for different types of cases and so on

There are some other things in the how-to and frequently asked question section of the documentation site as well

**Reply by antmar904** [reply]
Gotcha.  I just watched that video and Velociraptor is pretty cool plus it included KAPE! :)
